File: erpnext/selling/report/mineral_product_sales_report/mineral_product_sales_report.js
// Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt

frappe.query_reports["Mineral Product Sales Report"] = {
	"filters": [
		{
                        "fieldname": "from_date",
                        "label": __("From Date"),
                        "fieldtype": "Date",
                        "default": frappe.defaults.get_user_default("year_start_date"),
                },
                {
                        "fieldname": "to_date",
                        "label": __("To Date"),
                        "fieldtype": "Date",
                        "default": frappe.defaults.get_user_default("year_end_date"),
                },
		{
                        "fieldname": "sub_group",
                        "label": __("Item Sub Group"),
                        "fieldtype": "Select",
			//"defualt": run query report to get sub_group when item_group is Mineral Products
                        "options": ["Query Sand", "Drenched Sand", "RB Sand"]
                },
                {
                        "fieldname": "customer",
                        "label": __("Customer"),
                        "fieldtype": "Link",
                        "options": "Customer",
                }

	]
}

File: erpnext/selling/report/mineral_product_sales_report/mineral_product_sales_report.py
# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import _
def execute(filters=None):
	columns = get_columns()
	frappe.msgprint("{0}".format(len(columns)))
	data = 'tashi'
	#for i in xrange(19:
	#data.append(i)		
	return columns, data

def get_columns():
	return [
		{"fieldname": "item_code", "label": _("Item Code"), "fieldtype": "Link", "options":"Item", "width": 150},
		{"fieldname": "item_name", "label": _("Item Name"), "fieldtype": "Data",  "width": 150},
		{"fieldname": "customer_id", "label": _("Customer ID"), "fieldtype": "Link", "options":"Customer", "width": 150},
		{"fieldname": "customer_name", "label": _("Customer Name"), "fieldtype": "Data", "width": 150},
		{"fieldname": "customer_type", "label": _("Customer Type"), "fieldtype": "Data",  "width": 150},
		{"fieldname": "customer_cid", "label": _("Customer CID/Work Permit"), "fieldtype": "Data",  "width": 150},
		{"fieldname": "contact_no", "label": _("Contact No"), "fieldtype": "Data", "width": 150},
		{"fieldname": "plot_thram_no", "label": _("Plot/Thram No"), "fieldtype": "Data", "width": 150},
		{"fieldname": "location", "label": _("Destination"), "fieldtype": "Data", "width": 150},
		{"fieldname": "vehicle_no", "label": _("Vehicle No"), "fieldtype": "Data", "width": 150},
		{"fieldname": "vehicle_capacity", "label": _("Vehicle Capacity"), "fieldtype": "Data", "width": 150},
		{"fieldname": "driver_name", "label": _("Driver's Name"), "fieldtype": "Data", "width": 150},
		{"fieldname": "driver_contact_no", "label": _("Driver Contact No"), "fieldtype": "Data", "width": 150},
		{"fieldname": "current__address", "label": _("Current Address"), "fieldtype": "Data", "width": 150},
		{"fieldname": "qty_required", "label": _("Quantity Required(UOM)"), "fieldtype": "Data", "width": 150},
		{"fieldname": "qty_approved", "label": _("Quantity Appoved(UOM)"), "fieldtype": "Data", "width": 150},
		{"fieldname": "balance_qty", "label": _("Balanced Qty.(UOM)"), "fieldtype": "Data", "width": 150},
		{"fieldname": "rate", "label": _("Rate"), "fieldtype": "Data", "width": 150},
		{"fieldname": "amount", "label": _("Amount"), "fieldtype": "Data", "width": 150},

	]


